### PR TITLE
[bitnami/postgresql-ha] Add serviceAccount

### PR DESCRIPTION
**Pull request by erpel**
**Description of the change**

This change ports the mechanism to create a serviceAccount or use an existing serviceAccount from the bitnami/postgres chart.

**Benefits**

The postgresql-ha chart can be equipped with a dedicated serviceAccount, enabling the use of serviceAccounts with the required imagePullSecrets, permissions to use PodSecurityPolicies and other reasons to use serviceAccounts other than the default one.

**Possible drawbacks**

Since the feature is disabled by default and copies the already proven mechanism of the related bitnami/postgresql chart, the risk of introducing breaking changes with this is low.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

### CODE REVIEW

File: bitnami/postgresql-ha/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{{- if and (.Values.serviceAccount.enabled) (not .Values.serviceAccount.name) }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+  {{- include "common.labels.standard" . | nindent 4 }}
```
Comment (miguelaeh) [inline bitnami/postgresql-ha/templates/serviceaccount.yaml:6]:
```suggestion
  labels:  {{- include "common.labels.standard" . | nindent 4 }}
```

File: bitnami/postgresql-ha/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  name: {{ template "common.names.fullname" . }}
```
Comment (miguelaeh) [inline bitnami/postgresql-ha/templates/serviceaccount.yaml:7]:
```suggestion
  name: {{ include "common.names.fullname" . }}
```

File: bitnami/postgresql-ha/templates/serviceaccount.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+  namespace: {{ .Release.Namespace }}
```
Comment (miguelaeh) [inline bitnami/postgresql-ha/templates/serviceaccount.yaml:11]:
```suggestion
  namespace: {{ .Release.Namespace | quote }}
```

The quote is needed for certain characters on the namespace

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you very much for the PR!!!
Please take a look to some minor comments I left.

**Comment by erpel** [comment]
Should I just apply these changes to the bitnami/postgresql chart in this PR as well? It seems like a good thing to keep them in sync.

**Comment by miguelaeh** [comment]
Hi @erpel ,
I would create a different one and reference this one. Just to have one PR per chart. It is easier to review and to find the thread in case we need it in the future

**Review by miguelaeh — APPROVED** [review]
LGTM!

**Pull request merged**